sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/unit/tests/inputs/NewObjectsEmployeeInputFormat.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.beam.sdk.io.hadoop.inputformat.unit.tests.inputs;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.beam.sdk.io.hadoop.inputformat.HadoopInputFormatIO;
+import org.apache.beam.sdk.values.KV;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+/**
+ * <p>
+ * This is a valid InputFormat for reading employee data which is available in
+ * the form of {@code List<KV>} as
+ * {@linkplain NewObjectsEmployeeRecordReader#employeeDataList employeeDataList}. {@linkplain NewObjectsEmployeeRecordReader#employeeDataList
+ * employeeDataList} is populated using
+ * {@linkplain TestEmployeeDataSet#populateEmployeeData()}.
+ * <p>
+ * {@linkplain NewObjectsEmployeeInputFormat} splits data into
+ * {@value TestEmployeeDataSet#NUMBER_OF_SPLITS} splits, each split having
+ * {@value TestEmployeeDataSet#NUMBER_OF_RECORDS_IN_EACH_SPLIT} records each.
+ * {@linkplain NewObjectsEmployeeInputFormat} reads data from
+ * {@linkplain NewObjectsEmployeeRecordReader#employeeDataList employeeDataList}
+ * and produces a key (employee id) of type Text and value of type
+ * {@linkplain Employee Employee}.
+ * <p>
+ * {@linkplain NewObjectsEmployeeInputFormat} is also given as input to test
```

Comment:
To clarify this paragraph:
+ * {@linkplain NewObjectsEmployeeInputFormat} is used to test
 + * whether the {@linkplain HadoopInputFormatIO } source returns immutable records in
 + * the scenario when RecordReader creates new key and value objects every time it
 + * reads data.